cloudify_agent/api/pm/base.py

```diff
@@ -274,6 +287,22 @@ def __init__(self, logger=None, **params):
                               backend=self.broker_url)
         self._celery.conf.update(
             CELERY_TASK_RESULT_EXPIRES=defaults.CELERY_TASK_RESULT_EXPIRES)
+        if self._get_ssl_cert_path() != '':
+            self._celery.conf.BROKER_USE_SSL = {
+                'ca_certs': self._get_ssl_cert_path(),
+                'cert_reqs': ssl.CERT_REQUIRED,
+            }
+
+    def _create_celery_conf(self):
+        utils.render_template_to_file(
+            template_path='pm/shared/worker_conf.py.template',
+            file_path=os.path.join(self.virtualenv,
```

Comment:
It's creating the celery worker_conf.py, which is used to configure the SSL and the URL. The URL is easy to set outside of this (though if you see the comment in the template you'll note that the way we were doing it makes the credentials appear in a process listing). The SSL is a pain to set, as you really want to get the CERT_REQUIRED value in there, and trying to set it as a multiline variable in bash is doable but gets awkward fast.

As for moving it earlier in the lifecycle: I'm not sure we can- this is part of the agent install component anyway. When we're on the manager, this is easier to cope with as we can know where we will be putting agents. As soon as we start trying to deploy onto external nodes I don't think there's a better place for generating the required configs than this process manager module. The subclassed versions actually dropped some config in already (and indeed they had this call added to them).